nullaway/src/main/java/com/uber/nullaway/generics/CompareNullabilityVisitor.java

```diff
@@ -77,6 +81,9 @@ public Boolean visitClassType(Type.ClassType lhsType, Type rhsType) {
 
   @Override
   public Boolean visitArrayType(Type.ArrayType lhsType, Type rhsType) {
+    if (rhsType instanceof NullType) {
```

Comment:
@msridhar Since everywhere this visitor is called we have added a null check there is no way rhsType can ever be null at this point, so do we need this check?

Reply:
This is not checking if `rhsType` is null. It's checking if it's a `NullType` which is the type of `null`.